src/ufoLib2/objects/font.py

```diff
@@ -202,56 +204,72 @@ def read(cls, reader, lazy=True):
             self._reader = reader
         return self
 
-    def __contains__(self, name):
-        return name in self.layers.defaultLayer
-
-    def __delitem__(self, name):
-        del self.layers.defaultLayer[name]
-
-    def __getitem__(self, name):
-        return self.layers.defaultLayer[name]
-
-    def __setitem__(self, name, glyph):
-        self.layers.defaultLayer[name] = glyph
-
-    def __iter__(self):
-        return iter(self.layers.defaultLayer)
-
-    def __len__(self):
-        return len(self.layers.defaultLayer)
-
-    def get(self, name, default=None):
+    def __contains__(self, name: str) -> bool:
+        if self.layers.defaultLayer is not None:
+            return name in self.layers.defaultLayer
+        return False
+
+    def __delitem__(self, name: str) -> None:
+        if self.layers.defaultLayer is not None:
+            del self.layers.defaultLayer[name]
+        raise Error(f"No default layer present, cannot delete '{name}'.")
+
+    def __getitem__(self, name: str) -> Glyph:
+        if self.layers.defaultLayer is not None:
+            return self.layers.defaultLayer[name]
+        raise Error(f"No default layer present, cannot retrieve '{name}'.")
+
+    def __setitem__(self, name: str, glyph: Glyph) -> None:
+        if self.layers.defaultLayer is not None:
+            self.layers.defaultLayer[name] = glyph
+        raise Error(f"No default layer present, cannot insert '{name}'.")
+
+    def __iter__(self) -> Iterator[Glyph]:
+        if self.layers.defaultLayer is not None:
+            return iter(self.layers.defaultLayer)
+        raise Error("No default layer present, cannot iterate over it.")
+
+    def __len__(self) -> int:
+        if self.layers.defaultLayer is not None:
+            return len(self.layers.defaultLayer)
+        raise Error("No default layer present, cannot determine glyph count.")
+
+    def get(self, name: str, default: Optional[T] = None) -> Union[Optional[T], Glyph]:
         """Return the :class:`.Glyph` object for name if it is present in the
         default layer, otherwise return ``default``."""
-        return self.layers.defaultLayer.get(name, default)
+        if self.layers.defaultLayer is not None:
+            return self.layers.defaultLayer.get(name, default)
+        return default
```

Comment:
Same I would raise here. My point is: if you're writing a script using the font as a bag of glyphs, and the font doesn't have a default layer, then your script should fail, even if you're just checking whether something is in the font?